src/Button/Button.js

```diff
@@ -292,14 +292,26 @@ const Inner = createStyledComponent('span', styles.inner);
  */
 export default function Button({
   children,
+  element = 'button',
   iconStart,
   iconEnd,
   size = 'large',
   type = 'button',
   variant = 'regular',
   ...restProps
 }: Props) {
-  const rootProps = { size, text: children, type, variant, ...restProps };
+  const rootProps = {
+    size,
+    text: children,
+    type: 'button' === element ? type : undefined,
```

Comment:
I am a little bummed out that the `rootEl` being set to `element` doesn't automatically prevent the `type` attribute from being passed to the DOM element when `element` is `a`.  

In any case, I think this is probably better handled by using the `filterProps` Glamorous API.

```js
const Root = createStyledComponent(element, styles.button, {
  filterProps: element === 'button' ? undefined : ['type'],
  includeStyleReset: true,
  rootEl: element
});
```